src/smolagents/agents.py

```diff
@@ -755,6 +755,8 @@ def step(self, log_entry: ActionStep) -> Union[None, Any]:
                 tools_to_call_from=list(self.tools.values()),
                 stop_sequences=["Observation:"],
             )
+            if model_message.tool_calls is None or len(model_message.tool_calls) == 0:
+                raise Exception("Model did not call any tools. Call `final_answer` tool to return a final answer.")
```

Comment:
I am encountering this (trying to search via `DuckDuckGoSearchTool`). But I am not sure whether this is actually an error or expected application flow? (genuinely asking)

When debugging `model_message.content` sometimes looks like search results, but it might also be JSON like this:

```json
'{"call_id":"call_fa8d7xnj","result":{},"error_message":"","status":"OK","timestamp":1646238442}'
```